src/unstable-temp/DownloadImageModal/ImageForm.tsx

```diff
@@ -152,23 +157,42 @@ export const ImageForm = ({
 		);
 	}, [deviceType?.slug, rawVersion]);
 
+	const download = async (url: string) => {
+		setIsDownloadingType(DownloadTypeEnum.image);
+		try {
+			const res = await fetch(url, {
+				method: 'GET',
+				headers: new Headers({
+					Authorization: 'Bearer ' + authToken,
+				}),
+			});
+			const blob = await res.blob();
+			const newBlob = new Blob([blob]);
+
+			const blobUrl = window.URL.createObjectURL(newBlob);
+
+			const link = document.createElement('a');
+			link.href = blobUrl;
+			link.setAttribute('download', `${appId}.zip`);
+			document.body.appendChild(link);
+			link.click();
+			link.parentNode?.removeChild(link);
```

Comment:
Can't we say `document.body.removeChild(link)` instead of this since that's what we appended it to?

Reply:
This would remove the first 'a' element from the body and that's not what we want

Reply:
Ah okay